### PR TITLE
Match a number of functions and introduce anon unions

### PR DESCRIPTION
**Pull request by JoshDuMan**

### CODE REVIEW

**Comment by BowserSlug** [comment]
🚀🚀🚀🚀🚀🚀🚀 This PR matches 7 functions (+0.05%) on `us`.

⚠️ This PR introduces 34 warnings:
- include/common_structs.h:352: warning: unnamed struct/union that defines no instances
- include/common_structs.h:352: warning: unnamed struct/union that defines no instances
- include/common_structs.h:352: warning: unnamed struct/union that defines no instances
- include/common_structs.h:352: warning: unnamed struct/union that defines no instances
- include/common_structs.h:352: warning: unnamed struct/union that defines no instances
- include/common_structs.h:352: warning: unnamed struct/union that defines no instances
- include/common_structs.h:352: warning: unnamed struct/union that defines no instances
- include/common_structs.h:352: warning: unnamed struct/union that defines no instances
- include/common_structs.h:352: warning: unnamed struct/union that defines no instances
- include/common_structs.h:352: warning: unnamed struct/union that defines no instances
- include/common_structs.h:352: warning: unnamed struct/union that defines no instances
- include/common_structs.h:352: warning: unnamed struct/union that defines no instances
- include/common_structs.h:352: warning: unnamed struct/union that defines no instances
- include/common_structs.h:352: warning: unnamed struct/union that defines no instances
- include/common_structs.h:352: warning: unnamed struct/union that defines no instances
- include/common_structs.h:352: warning: unnamed struct/union that defines no instances
- include/common_structs.h:352: warning: unnamed struct/union that defines no instances
- include/common_structs.h:352: warning: unnamed struct/union that defines no instances
- include/common_structs.h:352: warning: unnamed struct/union that defines no instances
- include/common_structs.h:358: warning: unnamed struct/union that defines no instances
- include/common_structs.h:358: warning: unnamed struct/union that defines no instances
- include/common_structs.h:358: warning: unnamed struct/union that defines no instances
- include/common_structs.h:358: warning: unnamed struct/union that defines no instances
- include/common_structs.h:358: warning: unnamed struct/union that defines no instances
- include/common_structs.h:358: warning: unnamed struct/union that defines no instances
- include/common_structs.h:358: warning: unnamed struct/union that defines no instances
- include/common_structs.h:358: warning: unnamed struct/union that defines no instances
- include/common_structs.h:358: warning: unnamed struct/union that defines no instances
- include/common_structs.h:358: warning: unnamed struct/union that defines no instances
- include/common_structs.h:358: warning: unnamed struct/union that defines no instances
- include/common_structs.h:358: warning: unnamed struct/union that defines no instances
- include/common_structs.h:358: warning: unnamed struct/union that defines no instances
- include/common_structs.h:358: warning: unnamed struct/union that defines no instances
- include/common_structs.h:358: warning: unnamed struct/union that defines no instances
- include/common_structs.h:358: warning: unnamed struct/union that defines no instances
- include/common_structs.h:358: warning: unnamed struct/union that defines no instances
- include/common_structs.h:358: warning: unnamed struct/union that defines no instances
- include/common_structs.h:358: warning: unnamed struct/union that defines no instances
- src/196AA0.c:58: warning: implicit declaration of function `func_80268284'
- src/battle/partner/bow.c:35: warning: assignment from incompatible pointer type
- src/battle/partner/bow.c:82: warning: assignment from incompatible pointer type
- src/evt/cam_api.c:353: warning: initialization makes pointer from integer without a cast

**Comment by JoshDuMan** [comment]
These warnings are a result of the OS compiler not having anonymous unions. Ideally, these should be fixed by removing common.h from the OS files and replacing it with their proper header files.

File: src/196AA0.c [hunk 1]
```diff
@@ -90,98 +114,100 @@ void func_80268858(void) {
...
+            break;
+        default:
+            break;
     }
 }
-#else
-INCLUDE_ASM(s32, "196AA0", func_80268938);
-#endif
 
+void func_80268AF8(void);
```
Comment (ethteck) [inline src/196AA0.c:201]: If this is only needed in this file, could you put it above the first function?

File: src/196AA0.c [hunk 2]
```diff
@@ -90,98 +114,100 @@ void func_80268858(void) {
...
 INCLUDE_ASM(s32, "196AA0", func_80268AF8);
 
 INCLUDE_ASM(s32, "196AA0", func_80268C9C);
 
-INCLUDE_ASM(s32, "196AA0", func_80268E88);
+void func_80268E88(void) {
+    ActionCommandStatus *actionCmdStatus = &gActionCommandStatus;
```
Comment (ethteck) [inline src/196AA0.c:207]: Pointer on the left

File: src/196AA0.c [hunk 3]
```diff
@@ -369,9 +398,35 @@ ApiStatus func_802693F0(Evt* script, s32 isInitialCall) {
     return ApiStatus_DONE2;
 }
 
-INCLUDE_ASM(s32, "196AA0", CloseActionCommandInfo);
+ApiStatus CloseActionCommandInfo(Evt* script, s32 isInitialCall) {
+    ActionCommandStatus *actionCommandStatus = &gActionCommandStatus;
```
Comment (ethteck) [inline src/196AA0.c:402]: Pointer on the left

File: src/evt/cam_api.c [hunk 4]
```diff
@@ -189,7 +189,81 @@ ApiStatus SetCamTarget(Evt* script, s32 isInitialCall) {
...
+    f32 temp_f2;
+    f32 phi_f2;
+    s32 temp_a1;
+
+    if (isInitialCall) {
+        switch (temp_s3){
+            case 0:
+            case 1:
+                break;
+
```
Comment (ethteck) [inline src/evt/cam_api.c:208]: Unnecessary empty lines after break in this and case 2

File: src/evt/cam_api.c [hunk 5]
```diff
@@ -189,7 +189,81 @@ ApiStatus SetCamTarget(Evt* script, s32 isInitialCall) {
...
+            case 2:
+                temp_s4 *= 4;
+                break;
+
+            default:
+                break;
+        }
+
+        *(f32*)&script->functionTemp[3] = 1.0f;
```
Comment (ethteck) [inline src/evt/cam_api.c:217]: missed float grossness?

File: src/evt/cam_api.c [hunk 6]
```diff
@@ -189,7 +189,81 @@ ApiStatus SetCamTarget(Evt* script, s32 isInitialCall) {
...
+                script->functionTempF[3] = script->functionTempF[3] * -0.8;
+            } else {
+                script->functionTempF[3] = -script->functionTempF[3];
+            }
+            break;
+    }
+
+    if (script->functionTemp[1] == 0) {
+        camera->flags &= ~CAM_FLAG_SHAKING;
+        return 2;
```
Comment (ethteck) [inline src/evt/cam_api.c:262]: APIStatus consts (here and for 0 below)

**Review by ethteck — CHANGES_REQUESTED** [review]

File: src/evt/cam_api.c (hunk 5 above)
Comment (JoshDuMan) [reply src/evt/cam_api.c:217]: This one literally has to be like this. I can't get it otherwise.

Comment (ethteck) [reply src/evt/cam_api.c:215]: wait so storing it into a float version of the field doesn't produce the same code?

File: src/196AA0.c (hunk 1 above)
Comment (ethteck) [reply src/196AA0.c:201]: oh my bad. didn't see the include_asm below it. disregard!

File: src/evt/cam_api.c (hunk 5 above)
Comment (JoshDuMan) [reply src/evt/cam_api.c:215]: Correct, it changes when the save happens. I suspect there is something here with the switch above but I just couldn't get it to work.

**Pull request merged**